raft-tech/TANF-app
# Reparse Memory Management

## Summary of Changes
- Updated record deletion to use a paginator to avoid loading large quantities of records into memory
- Added  a little more logging for convenience
Pull request closes #3170 

## How to Test
_List the steps to test the PR_
These steps are generic, please adjust as necessary.
```
cd tdrs-frontend && docker-compose up
cd tdrs-backend && docker-compose up
```

1. Open http://localhost:3000/ and sign in.
2. Submit `ADS.E2J.NDM1.TS53_fake.txt` four or five times to get one million or more records into the DB. (This goes faster if you add more celery workers to `gunicorn_start.sh`!)
3. Exec into `web` and run `python manage.py clean_and_reparse -y 2023`. If you use the `-a` argument you actually won't see the issue occur because `-a` doesn't pass the queryset to Elastic DSL which is what causes it to come into memory.
4. Monitor the backend memory consumption while deleting records. It should increase, but only around 50-100MB.

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
+ [x] Clean and reparse command does not get killed (in general) when large querysets are involved.

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)



### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)


### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)


### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
